src/info.rs

```diff
@@ -27,6 +30,13 @@ impl From<[u8; 16]> for z_id_t {
     }
 }
 
+/// Formats the `z_id_t` into 16-digit hex string (LSB-first order)
+#[no_mangle]
+pub extern "C" fn z_id_to_string(zid: &z_id_t, dst: &mut MaybeUninit<z_owned_string_t>) {
```

Comment:
I'm maybe mistaken but aren't all our other functions are more like: `pub extern "C" fn z_id_to_string(dst: &mut MaybeUninit<z_owned_string_t>, zid: &z_id_t)` with the uninitialized memory as the first argument?

Reply:
I was following the deserialization API, but I'm open to this. 

```rust
pub unsafe extern "C" fn z_bytes_deserialize_into_string(
    this: &z_loaned_bytes_t,
    dst: &mut MaybeUninit<z_owned_string_t>,
) -> z_result_t {
```

Reply:
Depends on the logic. If the function is considered to be a constructor than MaybeUninit should be first, if the function is regarded as a class method (sort of ZId.to_string() in c++), than the first argument is z_id_t.

Reply:
This parameter order follows the same logic as `z_keyexpr_to_string` and `z_encoding_to_string` with desination in second parameter. So let's merge this variant to zenoh-c and update pico accordingly